COMPONENT-GUIDELINES.md

```diff
@@ -0,0 +1,293 @@
+# Purpose
+
+This document defines guidelines and patterns developers should follow when building components for the Pipedream registry.
+
+Developers may create, deploy and share components that do not conform to these guidelines, but they will not be eligible to be listed in the curated registry (e.g., they may be hosted in a Github repo). If you develop a component that does not adhere to these guidelines, but you believe there is value to the broader community, please [reach out in our community forum](https://pipedream.com/community). We can either provide guidance to improve compliance or evaluate if an exception is warranted.
+
+# Overview
+
+[Pipedream](https://pipedream.com) is a low code integration platform that makes it easy to connect APIs remarkably fast. Users can select from thousands of customizable, open source components for hundreds of apps, and orchestrate their execution in workflows. Developers can [contribute](CONTRIBUTING.md) to these open source components on [Github](https://github.com/pipedreamhq/pipedream) by:
+
+- Creating new comonents (sources and actions)
+- Updating exisiting components (e.g., fixing bugs, enhancing functionality)
+- Adding or updating metadata (e.g., descriptions, labels)
+
+Once a PR is merged to the `master` branch of the `pipedreamhq/pipedream` repo, the components are automatically registered and immediately become available to the 150k+ users of the Pipedream platform. 
+
+## Components
+
+Components are [Node.js modules](COMPONENT-API.md#component-structure) that run on Pipedream's serverless infrastructure. They may use Pipedream managed auth for 300+ apps and use most npm packages with no `npm install` or `package.json` required. Pipedream currently supports two types of components — sources and actions.
```

Comment:
Since this is a dev guide, it might be worth expanding on this section and highlighting some of the specific properties of components. e.g. when you talk about no `package.json` being required, I'm going to have lots of questions about what that means specifically, so might be worth linking to https://github.com/PipedreamHQ/pipedream/blob/master/COMPONENT-API.md#using-npm-packages .

Reply:
Like with components references, maybe link "sources" and "actions" to their sections below?